### PR TITLE
Deploy

### LINKED ISSUE #666: CLI: Add kusk deploy command
As a user I would like a convenient way to apply an OpenAPI spec with a simple command

This will also allow us to control the experience a bit and make it better for the user. 

`kusk deploy -i <openapispec>` should be the same as `kusk api generate -i <openapispec> | kubectl apply -f -`

While we want to add convenience, the end result should be identical between the two commands as it apply to the cluster changes.

### LINKED ISSUE #673: CLI: Kusk deploy with watch
As a user I want to have changes to an OpenAPI spec file auto-deployed to my cluster so that I have make quick iterative changes and see the results immediately.

Example: 
```
$ kusk deploy -i <spec file> --watch

Deployed ...
Watching file <spec file> for changes ...
Change detected ... redeploying ...
```

@donnalux, example above needs UX attention.

### PR DESCRIPTION
**Pull request by jasmingacic**
This PR 
closes #666 
closes #673 

See how it works:
https://user-images.githubusercontent.com/8327751/187888605-fcb1cadd-38f1-485c-9fc4-cb6503dbd8ab.mov

### CODE REVIEW

File: cmd/kusk/cmd/deploy.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/kubeshop/kusk-gateway/api/v1alpha1"
+	fileWatcher "github.com/kubeshop/kusk-gateway/cmd/kusk/internal/mocking/filewatcher"
```
Comment (mbana) [inline cmd/kusk/cmd/deploy.go:39]: Minor point, shouldn't import match the package name `filewatcher`.

File: cmd/kusk/cmd/deploy.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+func init() {
+	//add to root command
+	rootCmd.AddCommand(deployCmd)
+
+	deployCmd.Flags().StringVarP(&file, "file", "f", "", "file to deploy")
+	deployCmd.MarkFlagRequired("file")
+
+	deployCmd.Flags().BoolVarP(&watch, "watch", "w", false, "watch file changes and deploy on change")
+	deployCmd.Flags().StringVar(&name, "name", "", "name to name API with ")
```
Comment (mbana) [inline cmd/kusk/cmd/deploy.go:63]: `"name to name API with "`:

* Trailing space, and
* Perhaps better named, to me at least, "name of the API".

File: cmd/kusk/cmd/deploy.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+}
+
+// apiCmd represents the api command
+var deployCmd = &cobra.Command{
+	Use:   "deploy",
+	Short: "deploy command to deploy your apis",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
```
Comment (mbana) [inline cmd/kusk/cmd/deploy.go:72]: Up to you, if you want to change it, but there is also `RunE`, please see: https://github.com/spf13/cobra/blob/main/user_guide.md#returning-and-handling-errors.

File: cmd/kusk/cmd/deploy.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+		ctx := context.Background()
+		cmd.SilenceUsage = true
+		originalManifest, err := getParsedAndValidatedOpenAPISpec(file)
+		if err != nil {
```
Comment (mbana) [inline cmd/kusk/cmd/deploy.go:76]: Up to you, if you want to change it, but there is also `RunE`, please see: https://github.com/spf13/cobra/blob/main/user_guide.md#returning-and-handling-errors.

File: cmd/kusk/cmd/deploy.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+			fmt.Fprintln(os.Stderr, err)
+			os.Exit(1)
+		}
+		k8sclient, err := utils.GetK8sClient()
+		if err != nil {
```
Comment (mbana) [inline cmd/kusk/cmd/deploy.go:81]: Up to you, if you want to change it, but there is also `RunE`, please see: https://github.com/spf13/cobra/blob/main/user_guide.md#returning-and-handling-errors.

File: cmd/kusk/cmd/deploy.go [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+		}
+
+		api := &v1alpha1.API{}
+		yaml.Unmarshal([]byte(originalManifest), api)
+		if len(api.Namespace) == 0 {
+			api.Namespace = envoyFleetNamespace
+		}
+
+		if err := k8sclient.Create(ctx, api, &client.CreateOptions{}); err != nil {
+			fmt.Fprintln(os.Stderr, err)
```
Comment (mbana) [inline cmd/kusk/cmd/deploy.go:93]: Up to you, if you want to change it, but there is also `RunE`, please see: https://github.com/spf13/cobra/blob/main/user_guide.md#returning-and-handling-errors.

File: cmd/kusk/internal/utils/utils.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+	config, err := getConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return client.New(config, client.Options{Scheme: scheme})
+}
+
+func getConfig() (*rest.Config, error) {
```
Comment (mbana) [inline cmd/kusk/internal/utils/utils.go:29]: I think there might be some similar logic in `GetKubeConfig()` worth re-using.

**Review by mbana — DISMISSED** [review]
LGTM.

File: cmd/kusk/internal/utils/utils.go (hunk 7 above)
Comment (jasmingacic) [reply cmd/kusk/internal/utils/utils.go:29]: the one in package k8s returns string 
```
func GetKubeConfig() (string, error) {
```

**Review by mbana — APPROVED** [review]
LGTM.

**Pull request merged**